### PR TITLE
Make lib-core compile on VS2017

### PR DESCRIPTION
**Pull request by teams2ua**
* Update libraries
* Add googletest as submodule
* Adjust log sink implementations to new interface
* Minor include cleanings

### CODE REVIEW

**Comment by teams2ua** [comment]
Changes:
* Update in place fmt lib to 5.3.0, needed to do this as fmt 3.x use "unsigned" that not compiles in VS2017
* Update spdlog, needed to do this, because older version badly guard its bundled fmt version, so it starts to conflict with updated fmt lib.
* As a consequence to upper change need to update our implementation of spdlog sinks.
* Updated leveldb, bitcoin-core fork now have exactly the same changes Khalil did in our own branch, temporary it points to my fork until we merge https://github.com/LedgerHQ/leveldb/pull/3  we may want to drop our fork altogether and have CMakeLists.txt inside lib-core
* Made change to snappy
* Minor changes in a lot of files to make more consistent include path.

**Comment by phaazon** [comment]
@teams2ua the CI is still *dans les choux* 😞

**Comment by teams2ua** [comment]
It is building!

**Comment by phaazon** [comment]
Still that macos_release… What should we do about it?

File: core/src/debug/LogPrinterSink.cpp [hunk 1]
```diff
@@ -40,12 +41,15 @@ namespace ledger {
             set_level(spdlog::level::trace);
         }
 
-        void LogPrinterSink::log(const spdlog::details::log_msg &msg) {
+        void LogPrinterSink::sink_it_(const spdlog::details::log_msg &msg) {
             auto printer = _printer.lock();
             if (!printer)
                 return;
             auto level = msg.level;
-            auto message = msg.formatted.str();
```
[the remark this answers is not in the document]
Comment (teams2ua) [reply core/src/debug/LogPrinterSink.cpp]: In newer version of spdlog formatter is a part of logger, not a message, so here I just adapt the implementation.

File: core/src/debug/LogPrinterSink.hpp [hunk 2]
```diff
@@ -33,20 +33,24 @@
...
-        class LogPrinterSink : public spd::sinks::sink {
+        namespace api {
+            class LogPrinter;
+        };
+        class LogPrinterSink : public spd::sinks::base_sink<std::mutex> {
         public:
             LogPrinterSink(const std::shared_ptr<api::LogPrinter>& printer);
 
-            virtual void log(const spdlog::details::log_msg &msg) override;
+            virtual void sink_it_(const spdlog::details::log_msg &msg) override;
```
[the remark this answers is not in the document]
Comment (teams2ua) [reply core/src/debug/LogPrinterSink.hpp]: It is not me who create this name, I am just overriding.

File: core/src/debug/RotatingEncryptableSink.cpp [hunk 3]
```diff
@@ -56,46 +56,40 @@ namespace ledger {
...
-        void RotatingEncryptableSink::flush() {
+        void RotatingEncryptableSink::flush_() {
             auto context = _context;
             context->execute(make_runnable([this] () {
                _file_helper.flush();
             }));
         }
 
-        void RotatingEncryptableSink::_sink_it(std::string msg) {
+        void RotatingEncryptableSink::_sink_it(std::shared_ptr<fmt::memory_buffer> msg) {
```
[the remark this answers is not in the document]
Comment (teams2ua) [reply core/src/debug/RotatingEncryptableSink.cpp]: Changed interface ¯\_(ツ)_/¯

File: core/src/debug/RotatingEncryptableSink.cpp [hunk 4]
```diff
@@ -56,46 +56,40 @@ namespace ledger {
...
-            _file_helper.write(spdlog_msg);
+            _file_helper.write(*msg);
         }
 
         spdlog::filename_t RotatingEncryptableSink::calc_filename(std::shared_ptr<api::PathResolver> resolver,
                                                                   const spdlog::filename_t &filename, std::size_t index,
                                                                   const spdlog::filename_t &extension) {
-            std::conditional<std::is_same<spdlog::filename_t::value_type, char>::value, fmt::MemoryWriter, fmt::WMemoryWriter>::type w;
+            auto mangledFilename = fmt::format(SPDLOG_FILENAME_T("{}.{}"), filename, extension);
             if (index)
```
[the remark this answers is not in the document]
Comment (teams2ua) [reply core/src/debug/RotatingEncryptableSink.cpp]: Done

**Pull request merged**